packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -285,18 +290,95 @@ class NestedScrollView extends StatefulWidget {
   }
 
   @override
-  _NestedScrollViewState createState() => _NestedScrollViewState();
+  NestedScrollViewState createState() => NestedScrollViewState();
 }
 
-class _NestedScrollViewState extends State<NestedScrollView> {
+/// The [State] for a [NestedScrollView].
+///
+/// The [ScrollController]s, [innerController] and [outerController], of the
+/// [NestedScrollView]'s children may be accessed through its state. This is
+/// useful for obtaining respective scroll positions in the [NestedScrollView].
+///
+/// If you want to access the inner or outer scroll controller of a
+/// [NestedScrollView], you can get its [NestedScrollViewState] by supplying a
+/// `GlobalKey<NestedScrollViewState>` to the [NestedScrollView.key] parameter).
+/// The outer scroll position can also be obtained directly using
+/// [NestedScrollView.controller].
+///
+/// {@tool sample --template=stateless_widget_material}
+/// [NestedScrollViewState] can be obtained using a [GlobalKey].
+/// Using the following setup, you can access the inner scroll controller
+/// using `globalKey.currentState.innerController`.
+///
+/// ```dart preamble
+/// final GlobalKey<NestedScrollViewState> globalKey = GlobalKey();
+/// ```
+/// ```dart
+/// @override
+/// Widget build(BuildContext context) {
+///   return NestedScrollView(
+///     key: globalKey,
+///     headerSliverBuilder: (BuildContext context, bool innerBoxIsScrolled) {
+///       return <Widget>[
+///         SliverAppBar(
+///           title: Text('NestedScrollViewState Demo!'),
+///         ),
+///       ];
+///     },
+///     body: CustomScrollView(
+///       // Body slivers go here!
+///     ),
+///   );
+/// }
+///
+/// ScrollController get innerController {
+///   return globalKey.currentState.innerController;
+/// }
+/// ```
+/// {@end-tool}
+class NestedScrollViewState extends State<NestedScrollView> {
   final SliverOverlapAbsorberHandle _absorberHandle = SliverOverlapAbsorberHandle();
 
+  /// The [ScrollController] provided to the [ScrollView] in
+  /// [NestedScrollView.body].
+  ///
+  /// Manipulating the [ScrollPosition] of this controller pushes the outer
```

Comment:
No matter how I manipulate it, it will always push the header sliver(s) out of view? even if I scroll back all the way to the top?

Reply:
Yes. The inner controller can only be scrolled once the outer has reached the end of its scrollable. Therefore, if the inner is scrolling/being manipulated, the outer must be at the end its extent.